redwood-tooling-codegen/src/main/kotlin/app/cash/redwood/tooling/codegen/testingGeneration.kt

```diff
@@ -315,23 +310,18 @@ internal fun generateWidgetValue(schema: Schema, widget: Widget): FileSpec {
         defaultExpression = trait.defaultExpression?.let { CodeBlock.of(it) }
         addEqualsHashCodeToString(trait)
 
-        if (trait is ProtocolProperty) {
-          addToBuilder.addStatement(
-            "builder.changes += %T(widgetId, %T(%L), builder.json.%M(this.%N))",
-            Protocol.PropertyChange,
-            Protocol.PropertyTag,
-            trait.tag,
-            KotlinxSerialization.encodeToJsonElement,
-            trait.name,
-          )
-        }
+        toWidgetPropertiesBuilder.addStatement("instance.%1N(%1N)", trait.name)
       }
       is Children -> {
         type = Stdlib.List.parameterizedBy(RedwoodTesting.WidgetValue)
         defaultExpression = CodeBlock.of("%M()", Stdlib.listOf)
         addEqualsHashCodeToString(trait)
 
         childrenLists += CodeBlock.of("%N", trait.name)
+
+        toWidgetChildrenBuilder.beginControlFlow("for ((index, child) in %N.withIndex())", trait.name)
+          .addStatement("instance.%N.insert(index, child.toWidget(provider))", trait.name)
```

Comment:
This is the change that required #1110. We now access the children property in a loop. And while I could pull it into a local or use `apply` or `let` or something, still seems like a good change because it's a needless foot-gun otherwise.